### PR TITLE
Adds `CollectionDetailsPage` and `CollectionDetailsVM`

### PR DESCRIPTION
**Pull request by matuella**

### CODE REVIEW

File: lib/application/view-models/details/collection_details_vm.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  final CollectionServices collectionServices;
+  final ResourceServices resourceServices;
+
+  List<Resource>? _associatedResources;
+  late final StreamSubscription<CollectionStatus> _listener;
+
+  Future<void> _loadCollection() async {
+    final stream = await collectionServices.listenToCollectionStatus(collectionId: collectionId);
+
+    _listener = stream.listen((collectionStatus) async {
```
Comment (ggirotto) [inline lib/application/view-models/details/collection_details_vm.dart:53]: I couldn't understand why the details vm listen to the collection status. What's the scenario where the collection status may change while this page is loaded in memory?

**Review by ggirotto — CHANGES_REQUESTED** [review]

File: lib/application/view-models/details/collection_details_vm.dart (hunk 1 above)
Comment (matuella) [reply lib/application/view-models/details/collection_details_vm.dart:53]: the user might return to this page, for any possible reason, after executing the same collection

Comment (ggirotto) [reply lib/application/view-models/details/collection_details_vm.dart:53]: Ahh, got it. The execution is presented in a modal

**Review by ggirotto — APPROVED** [review]

**Comment by matuella** [comment]
All updates made and merged in the last PR (#76) in this nested chain of collection execution PRs

**Pull request closed without merging**